### ISSUE TITLE
go-fuse bindings fail to install in mingw64

### ISSUE THREAD

**Issue by fpqc** [dokan-dev/dokany#520]
Feature request can skip this form. Bug report **must** complete it. `Check List` must be 100% match or it will be automatically closed without further discussion. Please remove this line.

# Environment
 * Windows version:  Insiders Build 16199
 * Processor architecture:  amd64
 * Dokany version:  1.0.3
 * Library type (Dokany/FUSE): dokan_fuse mingw64 msys2

# Description

I compiled the mingw64 version of the dokan_fuse library in msys2, and then started to move to compile tmsu with FUSE.  The compilation instructions for the Linux version (which is FUSE-enabled, unlike the default Windows target) instruct one to install two go dependencies:

```
 go get -u github.com/mattn/go-sqlite3
 go get -u github.com/hanwen/go-fuse/fuse
```

However, the second command, which should install the fuse bindings for go, fails with the following error:
```
# github.com/hanwen/go-fuse/fuse
..\..\..\mingw64\src\github.com\hanwen\go-fuse\fuse\types.go:400: undefined: Attr
```
I installed dokan_fuse directly to my msys2/mingw64 environment (setting the cmake install prefix to /mingw64/), but it seems like the problem is with the go-fuse bindings is that compiling them requires one to have more than just libfuse.

Any idea why it might be failing to build?  I think this is more of a feature request than a bug report, since I read that building the go-fuse bindings fails on freebsd with the exact same error.

**Comment by fpqc** [comment]
There is another project called `go-fuse-c`, which wraps the C bindings, but build that fails like this:

```
$ go get -u github.com/vgough/go-fuse-c/fuse
# github.com/vgough/go-fuse-c/fuse
In file included from ..\..\..\mingw64\src\github.com\vgough\go-fuse-c\fuse\bridge.go:10:0:
./wrapper.h:11:56: fatal error: fuse/fuse_lowlevel.h: No such file or directory
 #include <fuse/fuse_lowlevel.h>  // IWYU pragma: export
```

**Comment by Liryna** [comment]
Hi @fpqc ,

Looks like `go-fuse-c` use `fuse_lowlevel` that is not available on dokan_fuse.
See @Rondom answer here for the reason that it is missing.
https://github.com/dokan-dev/dokany/pull/307#issuecomment-281628742

About go-fuse, error: https://github.com/hanwen/go-fuse/blob/master/fuse/types.go#L400
Seems to be that `Attr` is defined here https://github.com/hanwen/go-fuse/blob/fcb5b5b5c49286fb33e674534c18195d84abe8ef/fuse/types_linux.go#L15 but I don't know the project and cannot say why this is not used.

**Comment by fpqc** [comment]
@Liryna yeah, it looks like you'd have to implement the `go-fuse` project directly for dokan, since it doesn't have any dependencies on the C implementation libfuse.

Maybe writing a dokan go wrapper for go-fuse would be easier, I don't know.

**Comment by Liryna** [comment]
@fpqc There is a dokan go wrapper already. you can find the link of it in our wiki :)

**Comment by fpqc** [comment]
@Liryna does it implement dokan-fuse for go though?

**Comment by Liryna** [comment]
@fpqc Ohhh sorry my bad, no it will not wrapper the fuse call but use the dokan api. 

Should really look why fuse low level is used and if it is really required.

**Comment by cglong** [comment]
@Liryna Just as an aside, the link to the Go bindings is quite hidden. Might be good to promote that more 🙂 Thanks for mentioning it though!

**Comment by cglong** [comment]
In fact actually I'd recommend that the unofficial Go binding be promoted to an official one and moved to this repo alongside the Ruby, etc. ones. You could probably even give the same team R/W access to it and they can continue the majority of the development 🙂

**Comment by Liryna** [comment]
Hi @cglong ,

The go binding is linked with other wrappers in the wiki
https://github.com/dokan-dev/dokany/wiki/Dokan-API-Wrappers-language-compatible

I thought the same about you and proposed it on a ticket. You can try to reopen the subject if you want.
https://github.com/keybase/kbfs/issues/78

**Comment by Rondom** [comment]
@cglong Asking them whether they would be interested in moving their bindings to a separate repo, might under the dokan-dev-GitHub organisation is indeed a good idea. Not sure if they are interested, if not, we can always create a "dummy-repo" linking to code and documentation for better visibility (I understand that this was your main point, right?).

**Comment by cglong** [comment]
> we can always create a "dummy-repo" linking to code and documentation for better visibility (I understand that this was your main point, right?)

@Rondom, I think my main point was fundamentally that, when browsing the [Dokan homepage](https://dokan-dev.github.io/), there's no mention of Go bindings available. I think an update to the homepage or a dummy repo would be a good mitigation for my concerns 😄

**Comment by Liryna** [comment]
Hey thats a good idea !
Having at least the wrapper listed on the homepage with a link on it !

**Comment by taruti** [comment]
We should really have a separate repo for the code. I'll speak about it internally and try to make it happen during September. I'll comment on this when I know something.

**Comment by taruti** [comment]
Got the internal modifications done, pondering on the repo name,
https://github.com/keybase/dokan vs
https://github.com/keybase/dokan-go vs
https://github.com/keybase/go-dokan vs
...

**Comment by Liryna** [comment]
@taruti https://github.com/keybase/dokan-go would be perfect ❤️
